### PR TITLE
Update to clang 17 and re-enable clang-tidy checks

### PR DESCRIPTION
**Pull request by nsavoire**
# What does this PR do?

* Update clang to version 17
* Re-enable clang-tidy checks
* Fix the clang-tidy checks...

[ddprof-build PR](https://github.com/DataDog/ddprof-build/pull/112)
[Successful build](https://gitlab.ddbuild.io/DataDog/apm-reliability/ddprof/-/pipelines/20689102)

### CODE REVIEW

File: src/build_id.cc [hunk 1]
```diff
@@ -12,13 +12,11 @@ namespace ddprof {
 BuildIdStr format_build_id(BuildIdSpan build_id_span) {
   std::string build_id_str;
   // We encode every char on 2 hexa string characters
-  build_id_str.resize(build_id_span.size() * 2 + 1);
+  build_id_str.resize(build_id_span.size() * 2);
   for (unsigned i = 0; i < build_id_span.size(); ++i) {
-    snprintf(&build_id_str[static_cast<unsigned>(2 * i)], 3, "%02x",
-             static_cast<unsigned>(build_id_span[i]));
+    (void)snprintf(&build_id_str[static_cast<unsigned>(2 * i)], 3, "%02x",
```
Comment (r1viollet) [inline src/build_id.cc:17]: I don't understand why this would not do an invalid write
Will (2*build_id_span.size()) with an size of 3 not invalid write ?

Comment (r1viollet) [reply src/build_id.cc:17]: We talked about this. I ~agree that this is guaranteed to be OK by standard. Though it is still confusing.
I'll let you decide if you want to fix this.

Comment (nsavoire) [reply src/build_id.cc]: Changed the implementation to something simpler.

**Review by r1viollet — DISMISSED** [review]
LGTM

**Review by r1viollet — APPROVED** [review]
LGTM

**Pull request merged**